### ISSUE TITLE
Detach Process Bug

### ISSUE THREAD

**Issue by syjzwjj** [x64dbg/x64dbg#1850]
- Debugger version (Dec 24 2017 13:24:35).
- Operating system version and Service Pack (64 bits).

Seems the Detach function works incorrect. When the debugger stop it also terminate the debugee process. It shouldn't be like that.

**Comment by mrexodia** [comment]
duplicate of #648

**Comment by syjzwjj** [comment]
Hi~ Seems still not work , I use File->Detach , but the debugee still exist ... You can try , any svchost.exe process when you detach , it will be terminated.

**Comment by mrexodia** [comment]
That’s likely because you are attaching to a system process that doesn’t
like it. Try if the same happens on x64dbg itself...

**Comment by syjzwjj** [comment]
I tried a localservice privilege service like storage service. The command line is c:\windows\system32\svchost.exe -k localsystemnetworkrestricted -p -s StorSvc .  It still be terminated when I detach from x64dbg , but the debugger itself don't exist.

**Comment by mrexodia** [comment]
So just to confirm, if you do:

1. Open x64dbg twice
2. Use one x64dbg to attach to the other
3. File -> detach

It will exit x64dbg leaving you with one x64dbg.exe process?

**Comment by syjzwjj** [comment]
yes ,correct . If the process is an service process , the debugee will be terminated.You can try it ha.

**Comment by mrexodia** [comment]
I didn’t ask about a service process, I ask about a normal process...

**Comment by mrexodia** [comment]
And I cannot try it because I don’t have a computer near me :D

**Comment by syjzwjj** [comment]
Ohh. Normal process seems works fine . But due to some reason I need to debug some service process, so I found this issue by accident.

**Comment by Mattiwatti** [comment]
Debugging services is a PITA as they can't be launched the normal way through `CreateProcess` for several reasons, which is what it sounds like you're trying to do. See [Debugging a Service Application](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugging-a-service-application) for MS documentation on this (naturally focused on WinDbg, but much of it applies to x64dbg too). How painful it's going to be depends mostly on how early you want to start debugging the service. It's a lot easier to attach to a running process. Especially for `svchost.exe` services I would stay away from `Image File Execution Options` to 'auto-attach' a debugger, that's a nice way to make your system unbootable if you do it wrong.

With WinDbg you can do `windbg -psn LicenseManager` to isolate one service from others running in the same process and then attach. You'll still be debugging a running service though; if you need control from the first instruction the best way is a kernel debugger.

**Comment by blaquee** [comment]
You could always patch the Entrypoint of the Service executable (to EBFE) and make sure to adjust the timeout value in the registry and attempt to attach to the stalled Service entrypoint. Its a hack and it'll depend on how early the Service starts after system boot.

**Comment by G3ph4z** [comment]
> So just to confirm, if you do:
> 
> 1. Open x64dbg twice
> 2. Use one x64dbg to attach to the other
> 3. File -> detach
> 
> It will exit x64dbg leaving you with one x64dbg.exe process?
> […](#)

I tried to reproduce it this way, but I couldn't. I checked with other processes, such as Firefox and whatnow, but got the same results. Then, I tested it with svchost, yet I still couldn't get the described results.

I believe it can be closed.

**Comment by mrexodia** [comment]
Thanks!
